Lesson 2

amzhulanov opened the pull request.

amzhulanov committed:
Изменены текстовые сообщения
Исправлены методы преодоления препятствий

amzhulanov committed:
Изменены текстовые сообщения
Исправлены методы преодоления препятствий

amzhulanov committed:
Изменены текстовые сообщения
Исправлены методы преодоления препятствий

amzhulanov committed:
Изменены текстовые сообщения
Исправлены методы преодоления препятствий

amzhulanov committed:
Внедрен гендерный признак животного

amzhulanov committed:
Добавлены
метод вывода членов команды
метод вывода победителей

amzhulanov committed:
Исправлен состав команды добавлен вызов методов
ShowTeam
ShowResult

amzhulanov committed:
Сделал ДЗ, но вся обработка массивов происходит в одном классе

amzhulanov committed:
Сделал ДЗ, но вся обработка массивов происходит в одном классе

amzhulanov committed:
ДЗ выполнена

amzhulanov committed:
Delete Application.java

amzhulanov committed:
Delete Team.java

amzhulanov committed:
Delete Animal.java

amzhulanov committed:
Delete Cat.java

amzhulanov committed:
Delete Dog.java

amzhulanov committed:
Delete Human.java

amzhulanov committed:
Delete Robot.java

amzhulanov commented:
Пробовал работать с Git через Idea, не уверен, что всё правильно получилось.
Похоже я также коммитил файлы с первого ДЗ. Хоть я и нажал их удалить, в списке они остались.
Последние 4 файла относятся к ДЗ второго урока.

usharik commented on src/Java/ru/geekbrains/lesson2/CalcApplication.java:
```diff
@@ -0,0 +1,54 @@
+                ext=scn.nextInt();
+            }
+        }
+        while (ext==1);
```
Для контроля выхода из цикла лучше было бы создать отдельную логическую переменную. Не стоит так смешивать ответственности.

usharik commented on src/Java/ru/geekbrains/lesson2/EditArray.java:
```diff
@@ -0,0 +1,46 @@
+ public class EditArray {
+    static String[][] myArray=new String[4][4];
+
+    public static void writeArray(int dimension) {//метод для наполнения массива случайными числами и символом "а"
```
В этом задании гораздо удобнее объявлять массив при помощи литералов 
`String arr[][] = {{"1", "2", "3", "4"}, {"1", "2", "3", "4"}, {"1", "2", "3", "4"},{"1", "2", "3", "4"}}`

usharik commented on src/Java/ru/geekbrains/lesson2/EditArray.java:
```diff
@@ -0,0 +1,46 @@
+               }
+            }
+        }catch (NumberFormatException ex)
+                {throw new MyArrayDataException(String.format("Что-то пошло не так. В строке %s и столбце %s найден символ, отличный от числа. Программа завершает свою работу.",i+1,j+1));
```
В данном случае лучше передавать индексы в качестве параметров конструктора и конструировать сообщение внутри конструктора

usharik commented on src/Java/ru/geekbrains/lesson2/CalcApplication.java:
```diff
@@ -0,0 +1,54 @@
+            System.out.println("Выберите размерность заполнения массива от 1 до 4.");
+            int numMassiv = scn.nextInt();
+            if (numMassiv != 4) {
+                throw new MyArraySizeException("Размерность массива ограничена 4х4. Вы указали " + numMassiv+". Программа завершает свою работу.");
```
Размер массива нужно проверять не при вводе, а перед началом его обработки (вычислением суммы)

usharik commented on src/Java/ru/geekbrains/lesson2/CalcApplication.java:
```diff
@@ -0,0 +1,54 @@
+
+    public static void main(String[] args) throws MyArraySizeException, MyArrayDataException {
+        int ext=1;//переменная для выхода из цикла
+        try{Scanner scn=new Scanner(System.in);
```
Не приянто так оформлять код в блоках try

usharik commented on src/Java/ru/geekbrains/lesson2/CalcApplication.java:
```diff
@@ -0,0 +1,54 @@
+        while (ext==1);
+        } catch  (NumberFormatException ex) {
+            System.out.println("Ошибка преобразования числа " + ex+". Программа завершает свою работу.");
+        }catch (InputMismatchException ex){
```
Здесь должны перехватываться созданные вами исключения

amzhulanov committed:
Исключение MyArraySizeException перенёс в catch.
Старое местоположение просто закомментировал.

amzhulanov committed:
Merge remote-tracking branch 'origin/lesson-2' into lesson-2

amzhulanov committed:
ДЗ выполнена

amzhulanov committed:
ДЗ выполнена

amzhulanov committed:
Метод проверки наличия символа в массиве перенёс в основной класс.
В main оставил цикл для повтора программы и дополнительное исключение для контроля ввода ответа.
Размерность массива указывается в переменной dimensionArray

amzhulanov committed:
ДЗ выполнена

amzhulanov committed:
Изменил массив с public на private в классе CalcApplication

amzhulanov committed:
Чтобы массив был всегда разный, оставил его рандомное заполнение
Организовал передачу индексов в конструктор исключения для формирования конечной строки
Размер массива проверяется непосредственно перед его обработкой
